<?xml version="1.0"?>
<launch>
    <arg name="namespace"               default="lidar"     doc="{lidar, radar}" />
    <arg name="tracker"                 default="imm_ukf"   doc="{imm_ukf, cvkf}" />
    <arg name="tracker_input_topic"     default="objects" />
    <arg name="tracker_output_topic"    default="tracked_objects" />

    <group if="$(eval arg('namespace')=='lidar')">
        <group if="$(eval arg('tracker')=='imm_ukf')">
            <remap from="/detection/objects" to="$(arg tracker_input_topic)" />
            <remap from="objects" to="$(arg tracker_output_topic)" />

            <!--IMM UKF Tracker -->
            <node pkg="imm_ukf_pda_track" type="imm_ukf_pda" name="imm_ukf_pda_track" output="screen" />

            <!--Visualization -->
            <node pkg="detected_objects_visualizer" type="visualize_detected_objects" name="imm_ukf_pda_track_viz" output="screen" />
        </group>

        <group if="$(eval arg('tracker')=='cvkf')">

        <!--vectormap filtering -->
            <node pkg="autoware_ut" type="vectormap_filter" name="vectormap_filter" output="screen" />
            <remap from="input_objects" to="filtered_objects" />
            <remap from="objects" to="$(arg tracker_output_topic)" />

        <!--CVKF Tracker -->
            <node pkg="autoware_ut" type="cvkf_tracker.py" name="cvkf_track" output="screen"/>

        <!--Visualization -->
            <node pkg="detected_objects_visualizer" type="visualize_detected_objects" name="cvkf_track_viz" output="screen" />
        </group>
    </group>

    <group if="$(eval arg('namespace')=='radar')">
        <group if="$(eval arg('tracker')=='imm_ukf')">
            <remap from="/detection/objects" to="$(arg tracker_input_topic)" />
            <remap from="objects" to="$(arg tracker_output_topic)" />

            <!--IMM UKF Tracker -->
            <node pkg="imm_ukf_pda_track" type="imm_ukf_pda" name="imm_ukf_pda_track" output="screen" />

            <!--Visualization -->
            <node pkg="detected_objects_visualizer" type="visualize_detected_objects" name="imm_ukf_pda_track_viz" output="screen" />
        </group>

        <group if="$(eval arg('tracker')=='cvkf')">

            <!--vectormap filtering -->
            <node pkg="autoware_ut" type="vectormap_filter" name="vectormap_filter" output="screen" />

            <remap from="input_objects" to="filtered_objects" />
            <remap from="objects" to="$(arg tracker_output_topic)" />

            <!--CVKF Tracker -->
            <node pkg="autoware_ut" type="cvkf_tracker.py" name="cvkf_track" output="screen"/>

            <!--Visualization -->
            <node pkg="detected_objects_visualizer" type="visualize_detected_objects" name="cvkf_track_viz" output="screen" />
        </group>
    </group>

</launch>
